### PR TITLE
add working hour distribution for case study

### PR DESCRIPTION
**Pull request by gymgym1212**
#89 I copy the old working hour distribution part in `sql` and paste into `case-study/sqls/`. Like other sql component for case study, I add the clause `WHERE repo_id IN [{{repoIdList}}] ` between `FROM {{table}}` and `GROUP BY hour...`. In addition, I update the url in `postprocessor.js` and add a `"repoIdList": "[]"` in manifest.json file.

### CODE REVIEW

File: case-study/sqls/working-hour-distribution/sql [hunk 1]
```diff
@@ -0,0 +1 @@
+SELECT COUNT(*) AS count, toDayOfWeek(created_at) AS dayOfWeek, toHour(created_at) AS hour FROM {{table}} GROUP BY hour, dayOfWeek
```
Comment (frank-zsy) [inline case-study/sqls/working-hour-distribution/sql:1]: I didn't see `repoIdList` here and you need to give working hour distribution for every repo I think.

File: case-study/sqls/working-hour-distribution/image.svg [hunk 2]
```diff
@@ -0,0 +1,230 @@
+<svg
```
Comment (frank-zsy) [inline case-study/sqls/working-hour-distribution/image.svg:1]: You don't need to copy the image to case study folder, you can reuse this image in url.

File: case-study/sqls/working-hour-distribution/sql (hunk 1 above)
Comment (gymgym1212) [reply case-study/sqls/working-hour-distribution/sql:1]: Oops! My fault, and I forgot to commit the newest changes.

File: case-study/sqls/working-hour-distribution/image.svg (hunk 2 above)
Comment (gymgym1212) [reply case-study/sqls/working-hour-distribution/image.svg:1]: I got it!

File: case-study/sqls/working-hour-distribution/sql [hunk 3]
```diff
@@ -0,0 +1 @@
+SELECT COUNT(*) AS count, toDayOfWeek(created_at) AS dayOfWeek, toHour(created_at) AS hour FROM {{table}} WHERE repo_id IN [{{repoIdList}}] GROUP BY hour, dayOfWeek
```
Comment (frank-zsy) [inline case-study/sqls/working-hour-distribution/sql:1]: With this SQL you may get the working hour distribution for all repos in the case study, but I think we should have distribution for each repo.

File: case-study/sqls/working-hour-distribution/sql [hunk 4]
```diff
@@ -0,0 +1,12 @@
+SELECT 
+  COUNT(*) AS count, 
+  anyHeavy(repo_name) AS name,
+  anyHeavy(repo_id) as id,
+  toDayOfWeek(created_at) AS dayOfWeek, 
+  toHour(created_at) AS hour 
+FROM 
+  {{table}} 
+WHERE 
+  repo_id IN [{{repoIdList}}]
```
Comment (frank-zsy) [inline case-study/sqls/working-hour-distribution/sql:10]: By this SQL, you will select working hour of all the repos, but we may need to generate single working hour image for every repo in the case study.

Comment (gymgym1212) [reply case-study/sqls/working-hour-distribution/sql:10]: Yes, but in post-processor.js I distribute the result by repo_id and generate a list of svg links which have contained the <embed> tag.

Comment (frank-zsy) [reply case-study/sqls/working-hour-distribution/sql:10]: OK, my mistake, it looks fine here.

**Comment by frank-zsy** [comment]
And you will need to add the rendered text into at least one case study template file to make sure it will be rendered.

**Pull request merged**